src/main/kotlin/cacophony/codegen/patterns/cacophonyPatterns/PatternTemplates.kt

```diff
@@ -0,0 +1,31 @@
+package cacophony.codegen.patterns.cacophonyPatterns
+
+import cacophony.controlflow.CFGNode
+import cacophony.controlflow.RegisterLabel
+import cacophony.controlflow.ValueLabel
+
+abstract class RegisterAssignmentTemplate {
+    val lhsRegisterLabel = RegisterLabel()
+    val rhsLabel = ValueLabel()
+    val lhsSlot = CFGNode.RegisterSlot(lhsRegisterLabel)
+    val rhsSlot = CFGNode.ValueSlot(rhsLabel)
+}
+
+abstract class MemoryAssignmentTemplate {
+    val lhsLabel = ValueLabel()
+    val rhsLabel = ValueLabel()
+    val lhsSlot = CFGNode.MemoryAccess(CFGNode.ValueSlot(lhsLabel))
+    val rhsSlot = CFGNode.ValueSlot(rhsLabel)
+}
```

Comment:
Probably not a relevant comment for now, but something to keep in mind: when the number of patterns really grows, we could think of writing some code (maybe using reflection?) to generate matching pairs of instances for register and memory access patterns